### DISCUSSION TITLE
Running tasks from the root after package tasks complete

### DISCUSSION THREAD

**Discussion by mkvlrn-cm42** [vercel/turborepo#9027] [Help]
### Summary

I'm having a hard time understanding how to setup tasks in `turbo.json` in order to have the following happen when I run `turbo build`:

- all tasks named `build` in my packages run
- a task named `post-build` in the root `package.json` file run

This works fine, the tasks are executed perfectly in each package:
```json
{
  "$schema": "https://turbo.build/schema.json",
  "tasks": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": ["packages/*/dist/**"]
    }
  }
}

```

The `post-build` task in my root `package.json` will copy the outputs from the build task from each package to the root:
```jsonc
{
...
  "scripts": {
    "post-build": "yarn clean; yarn mv-shared; yarn mv-task-service; yarn mv-user-service",
    "clean": "rm -rf ./dist; mkdir dist",
    "mv-shared": "mv ./packages/shared/dist ./dist/shared",
    "mv-task-service": "mv ./packages/task-service/dist ./dist/task-service",
    "mv-user-service": "mv ./packages/user-service/dist ./dist/user-service"
  },
...
```

I know I could just add a script in the root with `turbo run build && yarn post-build`, and that's what I'm doing right now, but I would like to gradually add a more "native" turbo workflow to the project.

**Comment by anthonyshew** [comment]
It sounds like you could do `turbo run post-build` with a `turbo.json` of:

```json
{
  "tasks": {
    "build": {
      ...
    },
    "post-build": {
      "dependsOn": ["^build"]
    }
  }
}
```

You could do some script renaming here to make things a little more expressive of what you're actually meaning to do, but I haven't renamed above to help connect the logic of how it works more easily with where you're currently at.
